www/gatsby-browser.js

```diff
@@ -0,0 +1,9 @@
+exports.onRouteUpdate = (location) => {
+  const hash = location.hash || (location.location && location.location.hash)
```

Comment:
This is already setup in gatsby-link — try debugging that to see why it's not working? https://github.com/gatsbyjs/gatsby/blob/22e8755b1c0b178197cfeb7fa17ebf9f8839b4ac/packages/gatsby-link/src/index.js#L126